rust/processor/src/processors/econia_processor.rs

```diff
@@ -0,0 +1,200 @@
+use super::processor_trait::{ProcessingResult, ProcessorTrait};
+use crate::{
+    models::default_models::transactions::{TransactionDetail, TransactionModel},
+    utils::database::{execute_with_better_error, PgDbPool},
+};
+
+use aptos_indexer_protos::transaction::v1::Transaction;
+use async_trait::async_trait;
+use bigdecimal::BigDecimal;
+use chrono::{DateTime, Utc};
+use diesel::{result::Error, PgConnection};
+use econia_db::models::MarketRegistrationEvent;
+use econia_db::schema::market_registration_events;
+use serde_json::Value;
+use std::str::FromStr;
+use std::{collections::HashMap, fmt::Debug};
+
+pub const NAME: &str = "econia_processor";
+pub struct EconiaTransactionProcessor {
+    connection_pool: PgDbPool,
+    econia_address: String,
+}
+
+impl EconiaTransactionProcessor {
+    pub fn new(connection_pool: PgDbPool, econia_address: String) -> Self {
+        Self {
+            connection_pool,
+            econia_address,
+        }
+    }
+}
+
+impl Debug for EconiaTransactionProcessor {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let state = &self.connection_pool.state();
+        write!(
+            f,
+            "DefaultTransactionProcessor {{ connections: {:?}  idle_connections: {:?} }}",
+            state.connections, state.idle_connections
+        )
+    }
+}
+
+fn hex_to_string(hex: &str) -> String {
+    let hex_no_prefix = &hex[2..];
+    let hex_bytes = hex::decode(hex_no_prefix).unwrap();
+    String::from_utf8(hex_bytes).unwrap()
+}
```

Comment:
I think this should be a try from

Also there should be plenty of helper funcs throughout the Aptos types crate e.g. `AccountAddress`, then you can just use a method

Reply:
This isn't an account address, this is used for converting the module names and struct names from their utf-8 hex encoded format to a normal string. I don't know if there's already a utility for this and this code does the trick just fine.

I'd rather not "go fishing" for code that may/not exist when I can just write it myself.

Reply:
In Rust, traits can only be implemented on data structures defined in your crate. So `TryFrom` does not work here.

However, I would return `Result<String, FromHexError>` here instead, and remove the `unwrap`s.

In general, it is a good thing to not use `unwrap`s, as it will make the current thread panic (crash basically), and this will stop the program as well if the current thread is the main thread. Returning a result gives the possibility to the caller to handle the error.